stripe/http_client.py

```diff
@@ -206,6 +205,9 @@ def request(self, method, url, headers, post_data=None):
         if self._proxy:
             kwargs["proxies"] = self._proxy
 
+        if getattr(self._thread_local, "session", None) is None:
+            self._thread_local.session = requests.Session()
```

Comment:
Are we ignoring the `session` that was possibly passed to the constructor?

Reply:
Yeah, it might be a good idea to store the `session` parameter as `_session` on the instance, and then use it here if available like:

``` python
self._thread_local.session = _session or requests.Session()
```

Still has the problem of not being thread safe if used, but should be fine in most situations.